File: rock paper scissors.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

static int promptUserChoice(void) {
    char line[64];
    int choice;

    while (1) {
        printf("\nEnter your choice: ");
        if (!fgets(line, sizeof(line), stdin)) {
            return 0;
        }

        if (sscanf(line, "%d", &choice) == 1 && choice >= 1 && choice <= 3) {
            return choice;
        }

        printf("ERROR: Invalid choice! Please choose only 1, 2, or 3.\n");
    }
}

static char promptPlayAgain(void) {
    char line[64];

    while (1) {
        printf("\nDo you want to play another round? (Y/N): ");
        if (!fgets(line, sizeof(line), stdin)) {
            return 'N';
        }

        if (line[0] == 'Y' || line[0] == 'y' || line[0] == 'N' || line[0] == 'n') {
            return line[0];
        }

        printf("Invalid response. Please enter Y or N.\n");
    }
}

static const char *choiceName(int choice) {
    switch (choice) {
        case 1: return "Rock";
        case 2: return "Paper";
        default: return "Scissors";
    }
}

int main(void) {
    int userChoice, computerChoice;
    int userScore = 0, computerScore = 0;
    char playAgain;

    srand((unsigned int)time(NULL));

    printf("====================================\n");
    printf("       ROCK PAPER SCISSORS\n");
    printf("====================================\n");

    do {
        printf("\nChoose one of the following:\n");
        printf("1. Rock\n");
        printf("2. Paper\n");
        printf("3. Scissors\n");

        userChoice = promptUserChoice();
        if (userChoice == 0) {
            printf("\nInput error detected. Game ended.\n");
            break;
        }

        computerChoice = rand() % 3 + 1;

        printf("\nYou chose: %s\n", choiceName(userChoice));
        printf("Computer chose: %s\n", choiceName(computerChoice));

        if (userChoice == computerChoice) {
            printf("\nRESULT: It is a DRAW!\n");
        } else if ((userChoice == 1 && computerChoice == 3) ||
                   (userChoice == 2 && computerChoice == 1) ||
                   (userChoice == 3 && computerChoice == 2)) {
            printf("\nRESULT: YOU WIN THIS ROUND!\n");
            userScore++;
        } else {
            printf("\nRESULT: COMPUTER WINS THIS ROUND!\n");
            computerScore++;
        }

        printf("\n-----------------------------\n");
        printf("Your Score     : %d\n", userScore);
        printf("Computer Score : %d\n", computerScore);
        printf("-----------------------------\n");

        playAgain = promptPlayAgain();
        if (playAgain == 'Y' || playAgain == 'y') {
            printf("\nTRY AGAIN!\n");
        } else {
            printf("\nGame ended. Thank you for playing!\n");
        }

    } while (playAgain == 'Y' || playAgain == 'y');

    printf("\n====================================\n");
    printf("           FINAL SCORE\n");
    printf("====================================\n");
    printf("Your Score     : %d\n", userScore);
    printf("Computer Score : %d\n", computerScore);
    printf("====================================\n");

    if (userScore > computerScore) {
        printf("FINAL WINNER: YOU!\n");
    } else if (computerScore > userScore) {
        printf("FINAL WINNER: COMPUTER!\n");
    } else {
        printf("FINAL RESULT: DRAW!\n");
    }

    printf("====================================\n");
    return 0;
}

File: Rock_Paper_Scissors.java
import java.util.Random;
import java.util.Scanner;

public class Rock_Paper_Scissors {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);
        Random random = new Random();

        int userScore = 0;
        int computerScore = 0;
        int drawScore = 0;

        char playAgain = 'Y';

        System.out.println("====================================");
        System.out.println("       ROCK PAPER SCISSORS");
        System.out.println("====================================");

        while (playAgain == 'Y' || playAgain == 'y') {

            // Display options
            System.out.println("\nChoose one of the following:");
            System.out.println("1. Rock");
            System.out.println("2. Paper");
            System.out.println("3. Scissors");

            // Get user's choice
            System.out.print("\nEnter your choice: ");
            int userChoice = input.nextInt();

            // Validate user's choice
            if (userChoice < 1 || userChoice > 3) {
                System.out.println("\nERROR: Invalid choice!");
                System.out.println("Please choose only 1, 2, or 3.");
                System.out.println("Game ended.");
                break;
            }

            // Generate computer choice: 1, 2, or 3
            int computerChoice = random.nextInt(3) + 1;

            // Display user's choice
            System.out.print("\nYou chose: ");
            displayChoice(userChoice);

            // Display computer's choice
            System.out.print("Computer chose: ");
            displayChoice(computerChoice);

            // Compare choices
            if (userChoice == computerChoice) {

                System.out.println("\nRESULT: IT IS A DRAW!");
                drawScore++;

            } else if (
                    (userChoice == 1 && computerChoice == 3) ||
                    (userChoice == 2 && computerChoice == 1) ||
                    (userChoice == 3 && computerChoice == 2)
            ) {

                System.out.println("\nRESULT: YOU WIN THIS ROUND!");
                userScore++;

            } else {

                System.out.println("\nRESULT: COMPUTER WINS THIS ROUND!");
                computerScore++;
            }

            // Display current scores
            System.out.println("\n-----------------------------");
            System.out.println("Your Score     : " + userScore);
            System.out.println("Computer Score : " + computerScore);
            System.out.println("Draws          : " + drawScore);
            System.out.println("-----------------------------");

            // Ask if user wants another round
            System.out.print("\nDo you want to play another round? (Y/N): ");
            playAgain = input.next().charAt(0);

            if (playAgain == 'Y' || playAgain == 'y') {
                System.out.println("\nTRY AGAIN!");
            } else if (playAgain == 'N' || playAgain == 'n') {
                System.out.println("\nGame ended. Thank you for playing!");
            } else {
                System.out.println("\nERROR: Invalid response.");
                System.out.println("Game ended.");
                break;
            }
        }

        // Final score
        System.out.println("\n====================================");
        System.out.println("           FINAL SCORE");
        System.out.println("====================================");
        System.out.println("Your Score     : " + userScore);
        System.out.println("Computer Score : " + computerScore);
        System.out.println("Draws          : " + drawScore);
        System.out.println("====================================");

        // Overall winner
        if (userScore > computerScore) {
            System.out.println("FINAL WINNER: YOU!");
        } else if (computerScore > userScore) {
            System.out.println("FINAL WINNER: COMPUTER!");
        } else {
            System.out.println("FINAL RESULT: DRAW!");
        }

        System.out.println("====================================");

        input.close();
    }

    // Method to display the selected choice
    public static void displayChoice(int choice) {

        if (choice == 1) {
            System.out.println("Rock");
        } else if (choice == 2) {
            System.out.println("Paper");
        } else {
            System.out.println("Scissors");
        }
    }
}